### ISSUE TITLE
False positive for if-branch conversion

### ISSUE THREAD

**Issue by earthiverse** [kalessil/phpinspectionsea#2012]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 6.3.0 |
| Language level | PHP 8.3 |

Current behaviour
--
I have this function (which I use in tests for php middleware).
This plugin is telling me the `elseif` `Can be converted into if-branch (clearer intention, lower complexity numbers)`

```php
// array<string, mixed|callable(mixed): bool> $checks Array of attribute-value pairs to check.

function (ServerRequestInterface $request) use ($checks) {
    foreach ($checks as $attribute => $value) {
        $valueInRequest = $request->getAttribute($attribute);

        if (is_callable($value) && !$value($valueInRequest)) {
            return false;
        } elseif ($valueInRequest !== $value) {
            return false;
        }
    }
    return true;
}
```

Expected behaviour
--
I don't think the code it generates is equivalent, however. The "fix" would always return false if $value was a callable, and that callable returned true.

```php
function (ServerRequestInterface $request) use ($checks) {
    foreach ($checks as $attribute => $value) {
        $valueInRequest = $request->getAttribute($attribute);

        if (is_callable($value) && !$value($valueInRequest)) {
            return false;
        }
        if ($valueInRequest !== $value) {
            return false;
        }
    }
    return true;
}
```

**Comment by earthiverse** [comment]
No, this is me being a dumb-dumb.

There was a logic error in my original code, the `$value($valueInRequest)` check should have been nested

```php
if (is_callable($value)) {
    if (!$value($valueInRequest)) {
        return false;
    }
} elseif ($valueInRequest !== $value) {
    return false;
}
```
